### PR TITLE
Check aggregated signatures for block approvals

### PR DESCRIPTION
**Pull request by bowenwang1996**
Currently we don't check aggregated bls signature for block approvals. An adversary can exploit this to generate a valid block with very large weight.

### CODE REVIEW

File: chain/chain/src/test_utils.rs [hunk 1]
```diff
@@ -280,6 +280,30 @@ impl RuntimeAdapter for KeyValueRuntime {
         }
     }
 
+    fn verify_approval_signature(
```
Comment (ilblackdragon) [inline chain/chain/src/test_utils.rs]: hm, usually in this test util KeyValueRuntime we don't verify signatures. e.g. next function just returns true

File: chain/epoch_manager/src/lib.rs [hunk 2]
```diff
@@ -344,27 +344,39 @@ impl EpochManager {
...
+            if !validators.contains(&validator_stake.account_id) {
+                let is_slashed = slashed.contains(&validator_stake.account_id);
+                validators.insert(validator_stake.account_id.clone());
+                result.push((validator_stake, is_slashed));
             }
         }
         Ok(result)
     }
 
+    /// Returns all block producers in current epoch, with indicator is they are slashed or not.
```
Comment (ilblackdragon) [inline chain/epoch_manager/src/lib.rs]: "with indicator is they are slashed or not" -> "with indicator if they are slashed or not"

File: core/crypto/src/bls.rs [hunk 3]
```diff
@@ -215,15 +215,11 @@ impl BlsSignature {
     }
 
     pub fn verify_aggregate(&self, data: &[u8], public_keys: &[BlsPublicKey]) -> bool {
-        if !cfg!(feature = "fake_crypto") {
```
Comment (ilblackdragon) [inline core/crypto/src/bls.rs]: let's remove then feature from Cargo.toml's as well and above from verify_single.
And from using it in gitlab.yaml

File: chain/chain/src/test_utils.rs (hunk 1 above)
Comment (bowenwang1996) [reply chain/chain/src/test_utils.rs]: yeah but it is required for my test which expects the signature to fail

**Pull request merged**